### ISSUE TITLE
Sliding available dates

### ISSUE THREAD

**Issue by wotaen** [nathanreyes/v-calendar#67]
Hi,

I have a date picker with range selection. I want the user to be able to select max 10 days. So when the user starts dragging, it'll only allow him to select max 10 previous or 10 next days.

I wanted to achieve this with `available_dates` that take `start` and `end` from component's data attributes. Then in `drag` event of the calendar, I update the component's data attributes and I was hoping that only those 10 days will be enable.

However, as soon as I click on any date, the picker registers this as `input` event, no dragging is performed:

Here's the drag event
```
..
on_drag_datepicker: function(range) {
        this.range_limit_start = moment().tz(window.settings.timezone).subtract(9, 'days').toDate()
        this.range_limit_end =  moment().tz(window.settings.timezone).toDate()
      },
...
```

Here's `available_dates`:
```
available_dates: function () {
        return {
          start: this.range_limit_start,
          end: this.range_limit_end,
        }
      },
```

Any idea what am I doing wrong?

**Comment by nathanreyes** [comment]
I think you are on the right track. Here is a [jsfiddle](https://jsfiddle.net/nathanreyes/ew7vtn4r/) for how I would approach this.

Basically, watch the drag value. When it just starts (`val && !oldVal`), set your `availableDates` at that moment. Otherwise, when the drag clears (`!val`), clear the `availableDates`.

**Comment by nathanreyes** [comment]
Please advice if you have problems with this approach.

**Comment by wotaen** [comment]
Thanks for the suggestion. However I think there is still a problem and the bug needs to be reopened. It actually manifests in the same way as my original problem (starting comment). This [fiddle](https://jsfiddle.net/6u7y8hfp/) is your modified example. I've only added `@input` event. Do a selection. After it's selected (and alert shown), click on another date to start another selection...immediatelly the `input` event is fired, not possible to drag at this time.

**Comment by nathanreyes** [comment]
Ah ok. I do see the issue now.

**Comment by wotaen** [comment]
Is this fixed in 0.7.1? The js fiddle that I pasted earlier still suffers from the same issue (however I'm not sure if the CND version is the latest). However, I've updated to 0.7.1 in my project and I'm getting exactly same behavior as before

**Comment by nathanreyes** [comment]
Ok give it a shot now. 0.7.2 addresses an issue that came up when converting the date pickers to render functions.
